internal/backup_selector.go

```diff
@@ -17,10 +18,13 @@ const LatestString = "LATEST"
 // BackupSelector returns the name of storage backup chosen according to the internal rules.
 // Returns NoBackupsFoundError in case there are no backups matching the criteria.
 type BackupSelector interface {
-	Select(folder storage.Folder) (string, error)
+	// TODO (rodrijjke): Fix other usages
```

Comment:
we can switch to the `Select(folder storage.Folder) (Backup, error)` since the backup is no longer defined only by its name.